types/mongoose/index.d.ts

```diff
@@ -1144,7 +1144,7 @@ declare module "mongoose" {
 
   interface MongooseDocumentOptionals {
     /** The string version of this documents _id. */
-    id?: string;
+    id?: any;
```

Comment:
The documentation link says this is string?

Reply:
There is an `_id` field which is `any`, was your intention to use that?

Reply:
No, if you initialize a schema using option `{ id: false }`, the id property can be anything you want.

Reply:
I agree with you about `_id`, but not `id`. The documentation seems to indicate `id` is a string.

https://stackoverflow.com/questions/15724272/what-is-the-difference-between-id-and-id-in-mongoose

Reply:
Some user declare `id` in schema, like: `id: number`, it conflict.  so, I suggest mongoose use mongo `_id` directly or create itself `__id` or something like `__v`, not `id`.

Reply:
I asked in mongoose:
https://github.com/Automattic/mongoose/issues/5911

The `id` property is a mongoose getter than can be disabled. When disabled it becomes a regular mongoose document property which can be set to anything.
http://mongoosejs.com/docs/guide.html#id